src/main/java/com/exactpro/th2/crawler/state/RecoveryState.java

```diff
@@ -0,0 +1,300 @@
+/*
+ * Copyright 2021-2021 Exactpro (Exactpro Systems Limited)
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exactpro.th2.crawler.state;
+
+import com.exactpro.cradle.Direction;
+import com.exactpro.cradle.messages.StoredMessage;
+import com.exactpro.cradle.testevents.StoredTestEventWrapper;
+import com.exactpro.cradle.utils.CompressionUtils;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.SerializationFeature;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.time.Instant;
+import java.util.Map;
+import java.util.Optional;
+import java.util.StringJoiner;
+
+public class RecoveryState {
+    private final InnerEvent lastProcessedEvent;
+
+    private final Map<String, InnerMessage> lastProcessedMessages;
+
+    private final long lastNumberOfEvents;
+
+    private final long lastNumberOfMessages;
+
+    private static final ObjectMapper MAPPER = new ObjectMapper().findAndRegisterModules()
+            .configure(SerializationFeature.WRITE_DATES_AS_TIMESTAMPS, false);
+
+    private static final Logger logger = LoggerFactory.getLogger(RecoveryState.class);
+
+    public RecoveryState(@JsonProperty("lastProcessedEvent") InnerEvent lastProcessedEvent,
+                         @JsonProperty("lastProcessedMessages") Map<String, InnerMessage> lastProcessedMessages,
+                         @JsonProperty("lastNumberOfEvents") long lastNumberOfEvents,
+                         @JsonProperty("lastNumberOfMessages") long lastNumberOfMessages) {
+        this.lastProcessedEvent = Optional.ofNullable(lastProcessedEvent).orElse(null);
+        this.lastProcessedMessages = Optional.ofNullable(lastProcessedMessages).orElse(null);
```

Comment:
`lastNumberOfEvents` and `lastNumberOfMessages` cannot be `null` because they have a primitive type in the constructor